### PR TITLE
popups allow unicode

### PR DESCRIPTION
**Pull request by apatil**
Hi @ocefpaf, another small one. This allows unicode titles on popups.

### CODE REVIEW

**Comment by ocefpaf** [comment]
Hi @apatil we can do,
```python
isinstance(popup, (str, unicode))
```
to be shorter.  We also need to be smarter about this to ensure py3k compatibility.

Last but not least, we will probably need something like:
```python
'fancy string/unicode'.encode('ascii', 'xmlcharrefreplace')
```

To get a proper HTML.

Are you up to implement this?  If not, can you open an issue and reference this PR?

File: folium/folium.py [hunk 1]
```diff
@@ -616,7 +616,7 @@ def _popup_render(self, popup=None, mk_name=None, count=None,
         if not popup_on:
             return 'var no_pop = null;'
         else:
-            if isinstance(popup, str):
+            if isinstance(popup, str) or isinstance(popup, unicode):
                 popup_temp = self.env.get_template('simple_popup.js')
```
Comment (ocefpaf) [inline folium/folium.py]: Use `isinstance(popup, (str, unicode))` instead.

**Comment by apatil** [comment]
Hi @ocefpaf, that's a good test suite. :) I'll give it a shot later today.

**Comment by apatil** [comment]
How does that look, @ocefpaf?

**Comment by ocefpaf** [comment]
Looking great @apatil!  As a folium maintainer and a non-English speaker Thanks for doing this!!

I will review/test it and merge ASAP.

File: folium/folium.py [hunk 2]
```diff
@@ -616,10 +618,21 @@ def _popup_render(self, popup=None, mk_name=None, count=None,
...
+                if isinstance(popup, utype):
+                    popup_txt = popup.encode('ascii', 'xmlcharrefreplace')
+                else:
+                    popup_txt = popup
+                if sys.version_info >= (3,0):
+                    popup_txt = popup_txt.decode()
                 return popup_temp.render({'pop_name': mk_name + str(count),
-                                          'pop_txt': json.dumps(popup),
+                                          'pop_txt': json.dumps(str(popup_txt)),
```
Comment (ocefpaf) [inline folium/folium.py]: @apatil I am on the fence here.  I am not sure if the call to `str` is necessary, but it does not hurt...  What is your argument?

File: folium/folium.py [hunk 3]
```diff
@@ -653,6 +666,8 @@ def _popup_render(self, popup=None, mk_name=None, count=None,
                                           'max_width': max_width,
                                           'json_out': json_out,
                                           'vega_id': vega_id})
+            else:
+                raise TypeError("Unrecognized popup type")
```
Comment (ocefpaf) [inline folium/folium.py]: @apatil Can you change this to:
```python
raise TypeError("Unrecognized popup type: {!r}".format(popup))
```

File: folium/folium.py [hunk 4]
```diff
@@ -616,10 +618,21 @@ def _popup_render(self, popup=None, mk_name=None, count=None,
...
-            if isinstance(popup, str):
+            if sys.version_info >= (3,0):
+                utype, stype = str, bytes
+            else:
+                utype, stype = unicode, str
+
+            if isinstance(popup, (utype, stype)):
                 popup_temp = self.env.get_template('simple_popup.js')
+                if isinstance(popup, utype):
+                    popup_txt = popup.encode('ascii', 'xmlcharrefreplace')
```
Comment (ocefpaf) [inline folium/folium.py]: Loved it!

**Comment by ocefpaf** [comment]
@apatil I just finished review and it is looking really nice.  Can you address the my `raise` request and squash the commits?

Also, can you add an entry in the [Changelog](https://github.com/python-visualization/folium/blob/master/CHANGES.txt).

(If you do not have time for that let me know and I will merge and do it myself :wink: )

File: folium/folium.py (hunk 2 above)
Comment (apatil) [inline folium/folium.py]: I was getting type errors in python3 without it. The encode('ascii', ...) produces a bytes object, which json.dumps doesn't like.

Comment (ocefpaf) [inline folium/folium.py]: `str` it is then.

**Comment by apatil** [comment]
@ocefpaf sorry for the delay and the extra commit, but it's mostly squashed. :)

**Comment by ocefpaf** [comment]
Nice!  Thanks for sending another **very** useful PR!

**Comment by apatil** [comment]
My pleasure. :)

**Pull request merged**